From 34a77c1c5f4f868d7b1b14c0512b0976f7e45dd3 Mon Sep 17 00:00:00 2001
From: Joel Goddard <Joel.Goddard@arm.com>
Date: Thu, 21 Mar 2024 15:37:43 +0000
Subject: [PATCH 845/954] fix(mhu): use MHUv2 if PLAT_MHU_VERSION undefined

If RSS Comms is used but PLAT_MHU_VERSION was undefined then it should
default to MHUv2 to avoid breaking existing configurations which did not
need to specify PLAT_MHU_VERSION as on MHUv2 was available.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 69896b18dee6e941a261e2e65d7ed7982385bbdf fix(mhu): use MHUv2 if PLAT_MHU_VERSION undefined]
Change-Id: I8353b49b9f61414a664c2802f90ba3b2bc526887
Signed-off-by: Joel Goddard <joel.goddard@arm.com>
(cherry picked from commit c34dd06a843d71cdba2fa1c3c9067f6f130a0c73)
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/443837
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Tested-by: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 drivers/arm/rss/rss_comms.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/arm/rss/rss_comms.mk b/drivers/arm/rss/rss_comms.mk
index 91b830c2d2..0d1e308751 100644
--- a/drivers/arm/rss/rss_comms.mk
+++ b/drivers/arm/rss/rss_comms.mk
@@ -13,6 +13,9 @@ RSS_COMMS_SOURCES	:=	$(addprefix drivers/arm/rss/,			\
 					rss_comms_protocol_pointer_access.c	\
 				)
 
+# Default to MHUv2 if PLAT_MHU_VERSION undefined
+PLAT_MHU_VERSION ?= 2
+
 ifeq (${PLAT_MHU_VERSION}, 3)
 RSS_COMMS_SOURCES	+=	$(addprefix drivers/arm/mhu/,			\
 					mhu_v3_x.c				\
-- 
2.53.0

